modules/substrate/src/lib.rs

```diff
@@ -309,18 +329,44 @@ impl<T: Trait> BridgeStorage for PalletStorage<T> {
 	type Header = BridgedHeader<T>;
 
 	fn write_header(&mut self, header: &ImportedHeader<BridgedHeader<T>>) {
-		let hash = header.header.hash();
+		use core::cmp::Ordering;
+
+		let hash = header.hash();
+		let current_height = header.number();
+		let best_height = <ChainTipHeight<T>>::get();
+
+		match current_height.cmp(&best_height) {
```

Comment:
Q: what if we are finalizing header that has best height less than `ChainTipHeight`, but all headers in the `BestHeaders` collection are now abandoned (i.e. they have forked from finalized branch before best finalized header)

Reply:
Right now we'd continue to return the height of the abandoned fork as the best. I made a note about this in one of the `forks` tests saying that with pruning we'd be eventually consistent. Tomek didn't let that fly, so I gotta change it

Reply:
I agree with the other Tomek comment that it isn't super-critical now, so may be fixed later, in a separate issue